aoc2022/src/main/scala/day11.scala

```diff
@@ -0,0 +1,86 @@
+import scala.io.*
+import math.*
+import scala.annotation.tailrec
+
+/**
+ * PART 01:
+ *
+ * Involves following the instructions and implementing them. 
+ *
+ * PART 02:
+ *
+ * Uses the Chinese Remainder Theorem. Had to look that up based on a hint.
+ * 
+ * https://en.wikipedia.org/wiki/Chinese_remainder_theorem 
+ *
+ */
+
+
+object day11 extends App:
+
+  private val day: String =
+    this.getClass.getName.drop(3).init
+
+  private val start1: Long =
+    System.currentTimeMillis
+
+  private val input: Vector[Monkey] =
+
+    def parseMonkey(s: String): Monkey = s match
+      case s"Monkey$id:Startingitems:${items}Operation:${op}Test:divisibleby${test}Iftrue:throwtomonkey${t}Iffalse:throwtomonkey$f" =>
```

Comment:
This feels like a hacky way of parsing the Strings. Why not pattern match on the grouped Strings? This is from my code:
```
case Vector(s"Monkey $no:", s"  Starting items: $items", s"  Operation: new = old $op $i", s"  Test: divisible by $div", s"    If true: throw to monkey $trueTarget", s"    If false: throw to monkey $falseTarget")
```

Reply:
I looked at your code and I didn't even know that that was possible. Cool stuff! I updated my code based on this.